From: Russ Allbery <eagle@eyrie.org>
Date: Tue, 8 Apr 2014 18:28:33 -0700
Subject: Increase buffer size for timeout tests

Linux has apparently increased the amount of data it's willing to
buffer on network writes, so increase our data size to 8MB from 4MB
to force a timeout.

Change-Id: Iad69b496eece9b543aa77aa4ec9c2e1f98b6e06d
Reviewed-on: https://gerrit.stanford.edu/1477
Reviewed-by: Russ Allbery <rra@stanford.edu>
Tested-by: Russ Allbery <rra@stanford.edu>
---
 tests/util/network/client-t.c | 8 ++++----
 tests/util/tokens-t.c         | 8 ++++----
 2 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/tests/util/network/client-t.c b/tests/util/network/client-t.c
index 13bce09..aae31cd 100644
--- a/tests/util/network/client-t.c
+++ b/tests/util/network/client-t.c
@@ -5,7 +5,7 @@
  * which can be found at <http://www.eyrie.org/~eagle/software/rra-c-util/>.
  *
  * Written by Russ Allbery <eagle@eyrie.org>
- * Copyright 2005, 2013 Russ Allbery <eagle@eyrie.org>
+ * Copyright 2005, 2013, 2014 Russ Allbery <eagle@eyrie.org>
  * Copyright 2009, 2010, 2011, 2012, 2013
  *     The Board of Trustees of the Leland Stanford Junior University
  *
@@ -376,8 +376,8 @@ test_network_write(void)
     char *buffer;
 
     /* Create the data that we're going to send. */
-    buffer = bmalloc(4096 * 1024);
-    memset(buffer, 'a', 4096 * 1024);
+    buffer = bmalloc(8192 * 1024);
+    memset(buffer, 'a', 8192 * 1024);
 
     /* Create the listening socket. */
     fd = network_bind_ipv4(SOCK_STREAM, "127.0.0.1", 11119);
@@ -414,7 +414,7 @@ test_network_write(void)
      * A longer write cannot be completely absorbed before the client sleep,
      * so should fail with a timeout.
      */
-    ok(!network_write(c, buffer, 4096 * 1024, 1),
+    ok(!network_write(c, buffer, 8192 * 1024, 1),
        "network_write aborted with timeout");
     is_int(ETIMEDOUT, socket_errno, "...with correct error");
     alarm(0);
diff --git a/tests/util/tokens-t.c b/tests/util/tokens-t.c
index cbbac0e..5190b7c 100644
--- a/tests/util/tokens-t.c
+++ b/tests/util/tokens-t.c
@@ -2,7 +2,7 @@
  * tokens test suite.
  *
  * Written by Russ Allbery <eagle@eyrie.org>
- * Copyright 2006, 2007, 2009, 2010, 2012, 2013
+ * Copyright 2006, 2007, 2009, 2010, 2012, 2013, 2014
  *     The Board of Trustees of the Leland Stanford Junior University
  *
  * See LICENSE for licensing terms.
@@ -253,9 +253,9 @@ main(void)
         socket_close(server);
         exit(0);
     } else {
-        result.value = bmalloc(4096 * 1024);
-        memset(result.value, 'a', 4096 * 1024);
-        result.length = 4096 * 1024;
+        result.value = bmalloc(8192 * 1024);
+        memset(result.value, 'a', 8192 * 1024);
+        result.length = 8192 * 1024;
         client = create_client();
         status = token_send(client, 3, &result, 1);
         free(result.value);
